libcob/ChangeLog

```diff
@@ -1,4 +1,13 @@
 
+2023-11-29  Fabrice Le Fessant <fabrice.le_fessant@ocamlpro.com>
+
+	* common.h: export "cob_get_strerror" as a public function
+	* common.c: (cob_expand_env_string): use "getpid" instead
+	  of "cob_sys_getpid" to use the correct PID in case of "fork"
```

Comment:
Do you have an example where this is probematic? It isn't if you use `CBL_GC_FORK`, of course. Note that (especially with 4.x) there are also other elements that need to be adjusted.

Reply:
No, except the use of `fork()` in external C code.

Reply:
I'm not sure if/how we should handle that. The scenario's I think of are:

* "external" -> COBOL (then back, then fork then to COBOL in both)
* COBOL -> "external with fork" -> to COBOL

In both cases the runtime needs to know about that, because it needs to reset some things. While this is only the cached PID in GC 3.1+3.2, we may need more entries. For example trunk has improved locking with BDB and this is done internally by a "locker id" - and this may not be used in more than a single thread (or even process). fbdb keeps its locker ids and resets them (later acquiring a new one) for a cbl_sys_fork'd process.

So... _can_ we handle that outside of `cob_sys_fork`? I currently see no option.
If there is no option, why should we support a specific case of `fork` that may won't work otherwise?

It _seems_ better to explicit document that after the COBOL runtime is initialized you may only run `cob_sys_fork()` but not `fork()` directly. We can of course come up with an API function which we call from within `cob_sys_fork()` for childs and that external programs may use when they do a `fork()` before getting to COBOL again.

Opinions?